### PR TITLE
RealMPFR and ComplexMPC

### PR DESCRIPTION
**Pull request by isuruf**
This PR introduces `mpfr_class` a class to manage a `mpfr_t`, `RealMPFR` a class that extends `Number` and wraps the `mpfr_class` and also `mpc` counterparts.

Tests are to be added.

### CODE REVIEW

File: src/complex_mpc.cpp [hunk 1]
```diff
@@ -0,0 +1,572 @@
...
+bool ComplexMPC::__eq__(const Basic &o) const
+{
+    if (is_a<ComplexMPC>(o)) {
+        const ComplexMPC &s = static_cast<const ComplexMPC &>(o);
+        if (get_prec() == s.get_prec()) {
+            return mpc_cmp(this->i.get_mpc_t(), s.i.get_mpc_t()) == 0;
+        } else {
+            return false;
+        }
+    }
```
Comment (thilinarmtb) [inline src/complex_mpc.cpp]: I guess you can remove the block `31 - 33` ?

Comment (isuruf) [inline src/complex_mpc.cpp]: Yes. Will do that

**Comment by thilinarmtb** [comment]
Great job !! +1 to merge after adding tests.

**Comment by certik** [comment]
Looks very good, the `mpfr_class` and `mpc_class` are exactly as I thought they should be.

We should add tests for the new functionality, otherwise +1.

**Comment by certik** [comment]
@isuruf can you add tests, resolve conflicts and let's merge this?

**Comment by isuruf** [comment]
Added tests. Once #484 is in, I'll rebase again

**Comment by isuruf** [comment]
Added Python wrappers and a eval function `.n()`. Let me know whether it's okay.

**Comment by certik** [comment]
This looks good to me. +1 to merge if tests pass.

**Comment by certik** [comment]
The only thing missing are some Python tests for the wrappers, aren't they? I would add some as well.

**Comment by certik** [comment]
This looks good. Tests pass, merging. Can you send a new PR with some Python tests for this?

**Pull request merged**